src/realm/util/span.hpp

```diff
@@ -113,7 +113,7 @@ class Span {
     {
         REALM_ASSERT(extent == std::distance(begin, end));
     }
-#if 0 // VS 16.9 incorrectly rejects this. 16.10+ support it
+#if !defined(_MSC_VER) || _MSC_VER >= 1929 // Visual Studio 16.9 incorrectly rejects this. 16.10+ supports it
```

Comment:
This is unconditionally disabled because as long as we support compilers which don't support it we need to not be using it.

Reply:
Makes sense. I'll revert this change.